index.js

```diff
@@ -1,18 +1,112 @@
-const readline = require('readline');
-const rl = readline.createInterface(process.stdin, process.stdout);
+/*Limitation: Right now the program only checks for Y and H and assumes if it
+isn't either of those then it's N and L. Add a conditon to check for
+N and L then add what to do if a user adds something else */
+
+//Update: Added a while loop for input until a number is given.
+//Update: Pulled the lower limit, upper limit, and secret number code out into a single getNumber function.
+//Update: Added a check that the secretNumber was between the lower and upper limits
+
+/*To Do: Check if the cheat engine is working (check limits). Move the checks above the H/L code */
+
+//(?)Bug: If you pick the highest or lowest number and lie when it gets guessed, the program will guess a different number once
+
+/**********Boiler Plate Begin ************/
+const readline = require("readline");
+const readlineInterface = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
-    rl.question(questionText, resolve);
+    readlineInterface.question(questionText, resolve);
   });
 }
+/**********Boiler Plate End ************/
 
-start();
+//This function gets an input from the user, checks to make sure its an number and stores it as an int
+async function getNumber() {
+  let inputNumber = "inappropriate start value";
+  while (inputNumber !== parseInt(inputNumber)) {
+    let inputString = await ask("");
```

Comment:
Consider adding some sort of prompt text here for our poor human eyes and brains e.g. ">_"